tests/Factory/DbalFactory/DbalFactoryConfigureQueryLoggerTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Cycle\Tests\Factory\DbalFactory;
+
+use Psr\Container\NotFoundExceptionInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use RuntimeException;
+use stdClass;
+use Yiisoft\Aliases\Aliases;
+use Yiisoft\Yii\Cycle\Factory\DbalFactory;
+use Yiisoft\Yii\Cycle\Tests\Factory\DbalFactory\Stub\FakeDriver;
+
+final class DbalFactoryConfigureQueryLoggerTest extends BaseDbalFactoryTest
+{
+    /**
+     * @param string|LoggerInterface $logger Classname or object
+     * @return null|LoggerInterface
+     */
+    protected function prepareLoggerFromDbalFactory($logger): ?LoggerInterface
+    {
+        $factory = (new DbalFactory([
+            'query-logger' => $logger,
+            'default' => 'default',
+            'aliases' => [],
+            'databases' => [
+                'default' => ['connection' => 'fake']
+            ],
+            'connections' => [
+                'fake' => [
+                    'driver' => FakeDriver::class,
+                    'connection' => 'fake',
+                    'username' => '',
+                    'password' => '',
+                ]
+            ],
+        ]))($this->container);
+        return $factory->driver('fake')->getLogger();
+    }
+
+    public function testLoggerDefinitionAsStringDefinition(): void
+    {
+        $nullLogger = $this->container->get(NullLogger::class);
```

Comment:
Would be nice to use the LoggerInterface instead

Reply:
There won't be any difference for this test

Reply:
Yes of course, but if want to test behavior only you no need to use any implementations